src/traces/scattergl/index.js

```diff
@@ -886,17 +883,77 @@ function plot(container, subplot, cdata) {
 
             scene.fill2d.update(scene.fillOptions);
         }
-    }
 
-    // make sure selection layer is initialized if we require selection
-    var dragmode = layout.dragmode;
 
-    if(dragmode === 'lasso' || dragmode === 'select') {
-        if(scene.select2d && scene.selectBatch) {
-            scene.scatter2d.update(scene.unselectedOptions);
+        // update selection
```

Comment:
Separated code chunks are in single location now

Reply:
On the `gl2d_12` mock, calling `Plotly.restyle(gd, 'selected.marker.color', 'blue')` leads to:

![image](https://user-images.githubusercontent.com/6675409/35642368-ebba4958-0690-11e8-8a5b-22b7f53fb6d0.png)

Reply:
Should be fixed with 98cddf3